problem_ordering.md

```diff
@@ -0,0 +1,41 @@
+# Rust Track Problem Order
+
+The actual source of truth of problem order and topics is [config.json](config.json), but this file documents our reasoning behind the problem order in that file.
+
+## Background
+
+- https://github.com/exercism/xrust/issues/126
+- https://github.com/exercism/xrust/issues/127
+- http://designisrefactoring.com/2016/07/09/exercism-shouldnt-make-you-cry/
+
+## Our Approach
+
+We order our problems for students who are learning Rust. With that audience in mind, we follow two rules when adding new problems.
+
+- A exercise should introduce one new concept
+- Exercises should progress from Rust basics to advanced Rust concepts
+
+### One New Concept
+
+If a problem solution requires students to use both `Result` and `HashMap`, then at least one of these concepts should be introduced by an earlier problem.
+
+If the track has several problems that rely on a concept, group them together to reinforce learning.
+
+### Progress Through Rust
+
+#### Early Problems
+
+Early problems should focus on Rust syntax and concepts that are present in many programming languages: conditionals, looping, higher-order-functions.
```

Comment:
I think HOF will depend entirely on the students' background. Those coming from Ruby/JS/Python/Java 8 will be familiar. I think they are becoming pretty standard, though that might just be my years of Ruby speaking.